### ISSUE TITLE
Serialization is not implemented for type Nothing: while converting  to Nothing. (NOT_IMPLEMENTED)

### ISSUE THREAD

**Issue by den-crane** [ClickHouse/ClickHouse#43644]
```sql
CREATE TABLE processed_data
(
    `machine_id` UUID,
    `name` String,
    `timestamp` DateTime
)
ENGINE = ReplacingMergeTree
PARTITION BY toYYYYMM(timestamp)
ORDER BY (machine_id, name, timestamp);


insert into processed_data(machine_id, name, timestamp)
select generateUUIDv4(), toString(number), '2022-11-24 12:00:00'
from numbers(1e3);

SELECT
  toStartOfInterval(timestamp, INTERVAL 300 second) AS timestamp,
  SUM(1) / SUM(1) AS value 
FROM processed_data
WHERE timestamp > toDateTime('2022-11-24 00:19:00') AND timestamp <= toDateTime('2022-11-26 00:20:00')
GROUP BY timestamp 
ORDER BY timestamp   ;
┌───────────timestamp─┬─value─┐
│ 2022-11-24 12:00:00 │     1 │
└─────────────────────┴───────┘

SELECT
  toStartOfInterval(timestamp, INTERVAL 300 second) AS timestamp,
  SUM(1) / SUM(1) AS value 
FROM processed_data
WHERE timestamp > '2022-11-24 00:19:00' AND timestamp <= '2022-11-26 00:20:00'
GROUP BY timestamp 
ORDER BY timestamp ;

Received exception from server (version 22.11.1):
Code: 48. DB::Exception: Received from localhost:9000. DB::Exception: Serialization is not implemented for type Nothing: while converting '2022-11-24 00:19:00' to Nothing. (NOT_IMPLEMENTED)
(query: SELECT
  toStartOfInterval(timestamp, INTERVAL 300 second) AS timestamp,
  SUM(1) / SUM(1) AS value 
FROM processed_data
WHERE timestamp > '2022-11-24 00:19:00' AND timestamp <= '2022-11-26 00:20:00'
GROUP BY timestamp 
ORDER BY timestamp ;)
```

https://fiddle.clickhouse.com/d69cc41f-4e34-4a67-b6af-167de68123f5

**Comment by alex88** [comment]
FYI this issue seems to be fixed in 22.12 onwards (tested 22.12, 23.1)
https://fiddle.clickhouse.com/e0d847cd-8813-4a00-931f-1c7bcad80868

**Comment by rschu1ze** [comment]
Fixed indeed, added a test.
